### ISSUE TITLE
 Resize images and fixation on the same origin of the tumor

### ISSUE THREAD

**Issue by loseen01** [hms-dbmi/CHIEF#48]
I've resized my images, but I'm still having problems and I thought it might be because of the resizing process.
The image may have changed or become distorted. Is it possible for this to happen?
Also regarding the origin of tumor
When the following code allows, it can give me a different result on the same image in each run. Random seeds were used
The result has become fixed on all images. For example, that thyroid is showing on all origins (different images) , and if you change the seed number, it will also be fixed, but on another origin, for example breast, and so on.
This is the code I used
To predict the origin of tumor (before using the seed)
Does anyone have an idea why this is happening?

import torch
import torch.nn as nn
from torchvision import transforms
from PIL import Image
import os

from models.ctran import ctranspath
from Downstream.Tumor_origin.src.network import CHIEF_Tumor_origin

# **Increase PIL's MAX_IMAGE_PIXELS limit to handle large images**
# ⚠️ **Caution:** Disabling the pixel limit can expose your application to potential DoS attacks.
# Ensure that all input images are from trusted sources.
Image.MAX_IMAGE_PIXELS = None  # Disables the limit

# Constants
MEAN = (0.485, 0.456, 0.406)
STD = (0.229, 0.224, 0.225)

class TumorOriginPredictor:
    def __init__(self):
        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
        
        # Initialize feature extractor
        self.feature_extractor = self._init_feature_extractor()
        
        # Initialize classifier
        self.classifier = self._init_classifier()
        
        # Setup transforms
        self.transforms = transforms.Compose([
            transforms.Resize((224, 224)),  # Updated to resize to 224x224
            transforms.ToTensor(),
            transforms.Normalize(mean=MEAN, std=STD)
        ])
        
        # Label mapping
        self.label_dict = {
            0: 'Prostate', 1: 'Lung', 2: 'Endometrial', 3: 'Breast', 
            4: 'Head Neck', 5: 'Colorectal', 6: 'Thyroid', 7: 'Skin',
            8: 'Esophagogastric', 9: 'Ovarian', 10: 'Glioma', 11: 'Bladder',
            12: 'Adrenal', 13: 'Colon', 14: 'Germ Cell', 15: 'Pancreatobiliary',
            16: 'Liver', 17: 'Cervix'
        }

    def _init_feature_extractor(self):
        model = ctranspath()
        model.head = nn.Identity()
        
        # Load feature extractor weights
        weights = torch.load('./model_weight/CHIEF_CTransPath.pth', map_location=torch.device('cpu'))
        model.load_state_dict(weights['model'], strict=True)
        
        model = model.to(self.device)
        model.eval()
        return model
    
    def _init_classifier(self):
        model = CHIEF_Tumor_origin(n_classes=18)  # 18 tumor types
        
        # Load classifier weights
        weights = torch.load('./model_weight/CHIEF_finetune.pth', map_location=torch.device('cpu'))
        model.load_state_dict(weights, strict=False)
        
        model = model.to(self.device)
        model.eval()
        return model

    def predict(self, image_path):
        """
        Predict tumor origin from an image path
        
        Args:
            image_path (str): Path to the image file
            
        Returns:
            dict: Dictionary containing predicted class and probabilities
        """
        try:
            # Load and preprocess image
            image = Image.open(image_path)
            image = self.transforms(image).unsqueeze(0)
            image = image.to(self.device)
        except Image.DecompressionBombError as e:
            print(f"Error: The image is too large and may be a decompression bomb. Details: {e}")
            return {}
        except FileNotFoundError:
            print(f"Error: The image file '{image_path}' does not exist.")
            return {}
        except Exception as e:
            print(f"An unexpected error occurred while opening the image: {e}")
            return {}
        
        try:
            # Extract features
            with torch.no_grad():
                # Get patch features
                patch_features = self.feature_extractor(image)  # Shape: [1, 768]
                
                # Reshape features for classifier
                # The classifier expects features of shape [N, 768] where N is number of patches
                patch_features = patch_features.unsqueeze(0)  # Shape: [1, 1, 768]
                
                # Get predictions
                logits, probabilities = self.classifier(patch_features)
                
                # Get predicted class
                pred_class = torch.argmax(probabilities, dim=1).item()
                
                # Get class probabilities
                probs = probabilities.squeeze().cpu().numpy()
                
                # Create results dictionary
                results = {
                    'predicted_class': self.label_dict.get(pred_class, "Unknown"),
                    'confidence': float(probabilities[0][pred_class]),
                    'probabilities': {self.label_dict[i]: float(prob) 
                                    for i, prob in enumerate(probs)}
                }
                
            return results
        except Exception as e:
            print(f"Error during prediction: {str(e)}")
            return {}

def main():
    # Example usage
    predictor = TumorOriginPredictor()
    
    # Example image path
    image_path = "./exsample/a.tiff"
    
    if not os.path.exists(image_path):
        print(f"Error: Image file not found at {image_path}")
        return
        
    results = predictor.predict(image_path)
    
    if results:
        print("\nTumor Origin Prediction Results:")
        print(f"Predicted Class: {results['predicted_class']}")
        print(f"Confidence: {results['confidence']:.4f}")
        
        print("\nClass Probabilities:")
        for class_name, prob in results['probabilities'].items():
            print(f"{class_name}: {prob:.4f}")
    else:
        print("Prediction could not be completed due to errors.")

if __name__ == "__main__":
    main()
